src/wilson/dom.cljs

```diff
@@ -174,9 +175,19 @@
   (update elem 1 merge-attrs attrs))
 
 (defn with-class
-  "Adds a class to an element."
-  [cls elem]
-  (with-attrs {:class cls} elem))
+  "Adds a class to a component."
+  [cls component]
+  (let [parse-f2c (fn [render-fn & outer-props]
+                    (let [outer-comp (apply render-fn outer-props)]
+                      (fn [_ & inner-props]
+                        (let [inner-comp (apply outer-comp inner-props)]
+                          (with-attrs {:class cls} inner-comp)))))
+        [render-fn] (match component
+                      [render-fn & _]
+                      [render-fn])]
```

Comment:
match does not do anything useful here! you could just use `first`. If you wanted to use match appropriately, you should use [core.match with a guard][guard]. This would have the benefit that it will be easier to handle the cases with and without an attr map already present. Maybe this does that already; you should have a test for that :)

[guard]: https://github.com/clojure/core.match/wiki/Basic-usage#guards